Add support for badging to Safari TP

Th3S4mur41 opened the pull request:
<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->

#### Summary

Add support for badging to Safari Technology Preview

#### Test results and supporting details

https://webkit.org/blog/14112/badging-for-home-screen-web-apps/

#### Related issues

<!-- 🔨 If applicable, use "Fixes #XYZ" -->

<!-- ✅ After submitting, review the results of the "Checks" tab! -->

beidson commented:
Note:
Badging is not supported *in Safari* (on either macOS or iOS/iPadOS)
And therefore is not supported *in Safari Technology Preview*

Badging *is* supported for web apps saved to the Home Screen on iOS/iPadOS 16.4 and later.
And badging *is* supported for web apps saved to the Dock in Safari 17 on the macOS Sonoma beta.

Those user agents use WebKit, but are not "Safari"

Khaldaldwfan commented:
اريد تعلم

Th3S4mur41 commented:
@beidson thanks for the hint regarding the versions which wasn't quite clear from the announcement.
I updated the PR to mark the correct Safari version on macOS and added a note with a reference to Sonoma.

Now while, I understand that this is currently limited to apps added to the home screen / dock.
There was note about that the version 16.4 that was already referenced for iOS.
Should we add a note there too to make the limitation clear?

queengooborg commented on api/Navigator.json:
```diff
@@ -575,7 +575,8 @@
             "safari": {
-              "version_added": false
+              "version_added": "17",
+              "notes": "Badging is supported for web apps saved to the Dock in Safari 17 on the macOS Sonoma beta"
```
I assume that it's not just apps added to the dock, but any installed PWA (including in the Launchpad).  (If it really is just accessible for dock apps...well, that'd be pretty bizarre.  :P)  Let's simplify the note everywhere to the following:

```suggestion
              "notes": "Badging is supported for installed web apps on macOS Sonoma and higher."
```

Th3S4mur41 commented:
Thanks @queengooborg I just updated the note based on your suggestion
